rectools/metrics/coverage.py

```diff
@@ -0,0 +1,109 @@
+import attr
+import pandas as pd
+
+from rectools import Columns
+
+from .base import Catalog, MetricAtK
+
+
+@attr.s
+class ItemCoverage(MetricAtK):
+    """
+    Item space coverage (also referred as catalog coverage) is a metric that shows
+    what part of the items is covered by first k recommendations
+    ItemCoverage = #recommended_items / num_items
+
+    Parameters
+    ----------
+    k : int
+        Number of items in top of recommendations list that will be used to calculate metric.
+
+    """
+
+    def calc(self, reco: pd.DataFrame, catalog: Catalog) -> float:
+        """
+        Calculate item space coverage for all users
+
+        Parameters
+        ----------
+        reco : pd.DataFrame
+            Recommendations table with columns `Columns.User`, `Columns.Item`, `Columns.Rank`.
+        catalog : collection
+            Collection of unique item ids that could be used for recommendations.
+
+        Returns
+        -------
+        float
+            Value of metric.
+        """
+        reco_k_first_ranks = reco[reco[Columns.Rank] <= self.k]
+        return len(reco_k_first_ranks[Columns.Item].unique()) / len(catalog)
+
+    def calc_per_user(self, reco: pd.DataFrame, catalog: Catalog) -> pd.Series:
```

Comment:
Maybe for this metric `calc_per_user` method is meaningless since recommendations for user are unique and coverage depends only on `k` that we set up here